### ISSUE TITLE
CA1717 not fired on some non-english system formats (Polish, German)

### ISSUE THREAD

**Issue by hubuk** [dotnet/roslyn-analyzers#2229]
#### Analyzer package
[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version
v2.6.3

#### Diagnostic ID

[CA1717](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1717-only-flagsattribute-enums-should-have-plural-names)

#### Repro steps

The problem might also occur with different rules. CA1717 is the one that triggered described behavior in my case. Reproduction project is located here: https://github.com/hubuk/FxCopRepro
You may also create a simple project which violates CA1717 rule with enum like this:
```C#
public enum Fields
{
    Default = 0,
    First = 1
}
```

1. Use English (United States) system format:
1. Build the solution in Visual Studio or from dotnet CLI
1. Change system format to non-English
![image](https://user-images.githubusercontent.com/11485931/54676682-7e729600-4b01-11e9-8627-2ea2733a0299.png)
1. Kill all build related processes (dotnet.exe, msbuild.exe, etc) to make them use new format settings.
1. Build the solution in Visual Studio or from dotnet CLI

#### Expected behavior
There shall be a CA1717 warning reported in the last step: 
```
> dotnet msbuild /t:restore,rebuild .\FxCopProblem\FxCopProblem.csproj
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 32,95 ms for ...\FxCopProblem\FxCopProblem.csproj.
Fields.cs(5,17): warning CA1717: Only FlagsAttribute enums should have plural
names [...\FxCopProblem\FxCopProblem.csproj]
  FxCopProblem -> ...\FxCopProblem\bin\Debug\netstandard2.0\FxCopProblem.dll
```
#### Actual behavior
There is no warning reported in the last step, build succeeds: 
```
> dotnet msbuild /t:restore,rebuild .\FxCopProblem\FxCopProblem.csproj
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 32,95 ms for ...\FxCopProblem\FxCopProblem.csproj.
  FxCopProblem -> ...\FxCopProblem\bin\Debug\netstandard2.0\FxCopProblem.dll
```

**Comment by genlu** [comment]
@hubuk Thanks for providing feedback! This behavior is by design, to be consistent with the old FxCop behavior.

https://github.com/dotnet/roslyn-analyzers/blob/3bb0de8da1b17cdd2f8e39af52510a6083f1abff/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumsShouldHavePluralNames.cs#L99

**Comment by hubuk** [comment]
@genlu Any plans to support other cultures? Would you accept pull-request for it? I am not familiar with the code, but it is probable that removal of this condition will not introduce any regression.
This behavior is wrong for many reasons:
1. There is no message that informs the user that the rule has not been executed for the code. 
1. It creates a false impression that code is OK when it isn't.
1. The culture seems to be irrelevant for this rule.
1. I couldn't find any documentation for this behavior.
1. It is really hard to diagnose. It took me almost a day and forced me to reinstall entire Visual Studio to see why it is failing on a build server and not failing on my dev environment at the same time.
1. Is seems like this is the only rule with such limitation.
1. I see no reason why keeping this consistent with old FxCop is beneficial to anyone.

**Comment by artberri** [comment]
I absolutely agree with @hubuk Having different behaviors based on region configuration makes not sense for me.

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/2809
